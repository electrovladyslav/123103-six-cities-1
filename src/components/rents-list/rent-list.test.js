import React from 'react';
import renderer from 'react-test-renderer';

import RentList from './rents-list';

const mockState = {
  offers: [],
  cityName: `Bangkok`,
  rentsCount: 2000,
  onCardTitleClick: jest.fn(),
  onElementActivate: jest.fn(),
};

it(`Rent list component renders correctly`, () => {
  const tree = renderer
  .create(<RentList
    elements ={mockState.offers}
    cityName ={mockState.cityName}
    rentsCount ={mockState.rentsCount}
    onCardTitleClick = {mockState.onCardTitleClick}
    onElementActivate = {mockState.onElementActivate}
  />)
  .toJSON();
  expect(tree).toMatchSnapshot();
});
