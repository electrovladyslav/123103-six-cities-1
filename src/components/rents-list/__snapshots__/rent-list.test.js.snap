// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rent list component renders correctly 1`] = `
<section
  className="cities__places places"
>
  <h2
    className="visually-hidden"
  >
    Places
  </h2>
  <b
    className="places__found"
  >
    2000
     places to stay in 
    Bangkok
  </b>
  <form
    action="#"
    className="places__sorting"
    method="get"
  >
    <span
      className="places__sorting-caption"
    >
      Sort by
    </span>
    <span
      className="places__sorting-type"
      tabIndex="0"
    >
      Popular
      <svg
        className="places__sorting-arrow"
        height="4"
        width="7"
      >
        <use
          xlinkHref="#icon-arrow-select"
        />
      </svg>
    </span>
    <ul
      className="places__options places__options--custom places__options--opened"
    >
      <li
        className="places__option places__option--active"
        tabIndex="0"
      >
        Popular
      </li>
      <li
        className="places__option"
        tabIndex="0"
      >
        Price: low to high
      </li>
      <li
        className="places__option"
        tabIndex="0"
      >
        Price: high to low
      </li>
      <li
        className="places__option"
        tabIndex="0"
      >
        Top rated first
      </li>
    </ul>
  </form>
  <div
    className="cities__places-list places__list tabs__content"
  />
</section>
`;
